test/e2e/deployment.go

```diff
@@ -374,17 +374,31 @@ func (d *Deployment) WaitForEnvoyUpdated() error {
 }
 
 func (d *Deployment) waitForEnvoyDaemonSetUpdated() error {
-	daemonSetUpdated := func() (bool, error) {
-		tempDS := new(apps_v1.DaemonSet)
-		if err := d.client.Get(context.TODO(), client.ObjectKeyFromObject(d.EnvoyDaemonSet), tempDS); err != nil {
+	contourPodImage := d.ContourDeployment.Spec.Template.Spec.Containers[0].Image
+	updatedPods := func() (bool, error) {
+		pods := new(v1.PodList)
+		labelSelectAppContour := &client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(d.EnvoyDaemonSet.Spec.Selector.MatchLabels),
+			Namespace:     d.EnvoyDaemonSet.Namespace,
+		}
+		if err := d.client.List(context.TODO(), pods, labelSelectAppContour); err != nil {
 			return false, err
 		}
-		return tempDS.Status.ObservedGeneration == tempDS.Generation &&
-			tempDS.Status.NumberAvailable > 0 &&
-			tempDS.Status.NumberAvailable == tempDS.Status.DesiredNumberScheduled &&
-			tempDS.Status.UpdatedNumberScheduled == tempDS.Status.DesiredNumberScheduled, nil
+
+		updatedPods := 0
+		for _, pod := range pods.Items {
+			if pod.Spec.InitContainers[0].Image != contourPodImage {
```

Comment:
yeah this makes sense as the indicator to ensure the pods actually have been updated with the new contour

Reply:
ah, looks like we're getting a panic here since we don't have an init container in many of our e2e test suites: https://github.com/projectcontour/contour/actions/runs/4067086318/jobs/7004630297#step:7:160

see: https://github.com/projectcontour/contour/blob/052a12d88decdb068926333b9f90ae77cb650a5c/test/e2e/deployment.go#L557 (we remove the init container in that case since we just run `contour bootstrap` locally)